docs/tutorials/nodes-tutorial.ipynb

```diff
@@ -0,0 +1,376 @@
+   "source": [
+    "pp.show_graph(a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b46191da-9516-402f-8f01-ad86014d8bc9",
+   "metadata": {},
+   "source": [
+    "And the resulting figure displays the smoothed data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1c986694-254e-4b42-a3f0-ae0d1408a3c1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4b69e6aa-88ad-419b-b9af-acd581645e8c",
+   "metadata": {},
+   "source": [
+    "## Adding interactive widgets\n",
+    "\n",
+    "In the example above, the kernel size `sigma` for the gaussian smoothing was frozen to `5`.\n",
+    "But we would actually want to control this via a slider widget.\n",
+    "\n",
+    "In this case, the smoothing node `b` now needs two inputs: the raw data, and the `sigma`.\n",
+    "It gets the raw data from the `input_node`, and the `sigma` from a `widget_node`,\n",
+    "which is coupled to a slider from the `ipywidgets` library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c483edcd-f28c-48c7-a6f4-958964a03bc8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = pp.input_node(da)\n",
+    "\n",
+    "import ipywidgets as ipw\n",
+    "slider = ipw.IntSlider(min=1, max=10)\n",
+    "b = pp.widget_node(slider)\n",
+    "\n",
+    "c = pp.node(gaussian_filter)(a, sigma=b)\n",
+    "\n",
+    "fig = pp.figure2d(c)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c671a291-bf40-40f7-9e03-75dbd77997b2",
+   "metadata": {},
+   "source": [
+    "As expected, the smoothing node `c` now has two parent nodes:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "743c2fdb-9782-439e-9e25-e3afe5960ad3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pp.show_graph(a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "08ce4da8-21ea-4aa4-bedc-4b38f01fdcdd",
+   "metadata": {},
+   "source": [
+    "And we can display the figure and the slider inside the same container:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9d35d552-af80-46f4-82b3-638d85a2b40a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ipw.VBox([slider, fig])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5dd33c3d-1e0b-4989-b0f2-1c1377364312",
+   "metadata": {},
+   "source": [
+    "When something changes in one of the nodes, all the nodes below it in the graph are notified about the change (the children nodes receive a notification, and they, in turn, notify their own children).\n",
+    "It is then up to each view to decide whether they are interested in the notification or not (usually, most views are interested in all notifications from parents).\n",
+    "If they are, they request data from their parent nodes, which in turn request data from their parents, and so on, until the request has reached the top of the graph.\n",
```

Comment:
This was explained earlier. Do we need it twice?

Reply:
I made the first explanation shorter so that it feels like the details aren't given twice.